src/main/java/randoop/condition/SpecificationTranslator.java

```diff
@@ -0,0 +1,388 @@
+package randoop.condition;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import randoop.compile.SequenceCompiler;
+import randoop.condition.specification.Guard;
+import randoop.condition.specification.Identifiers;
+import randoop.condition.specification.OperationSpecification;
+import randoop.condition.specification.Postcondition;
+import randoop.condition.specification.Precondition;
+import randoop.condition.specification.Property;
+import randoop.condition.specification.ThrowsCondition;
+import randoop.main.GenInputsAbstract;
+import randoop.reflection.RawSignature;
+import randoop.types.ClassOrInterfaceType;
+import randoop.util.Log;
+
+/** Translates an {@link OperationSpecification} object to an {@link OperationConditions} object. */
+public class SpecificationTranslator {
+
+  /** The name of dummy variables used by {@link randoop.contract.ObjectContract}. */
+  private static final String DUMMY_VARIABLE_NAME = "x";
+
+  /** The {@link RawSignature} for a guard expression method. */
+  private RawSignature guardExpressionSignature;
+
+  /** The parameter declaration string, with parentheses, for a guard expression method. */
+  private final String guardExpressionDeclaration;
```

Comment:
State the relationship between the RawSignature and this string.  Does one contain variable names?
(Then comments also apply to the `property*` variables.)